Sources/SpelledPitch/PitchSpeller/Wetherfield/FlowNetwork.swift

```diff
@@ -35,44 +35,55 @@ extension FlowNetwork {
 }
 
 extension WeightedDirectedGraph where Weight: Comparable {
-    // TODO: Make throw
-    mutating func reduceFlow(through edge: Edge, by amount: Weight) {
-        updateEdge(edge) { weight in weight - amount }
-    }
 
-    /// Removes the given edge if its weight is `0`. This happens after an edge, which has the
-    /// minimum flow of an augmenting path, is reduced by the minimum flow (which is its previous
-    /// value).
-    mutating func removeEdgeIfFlowless(_ edge: Edge) {
-        if weight(edge) == 0 {
-            removeEdge(edge)
+    /// Signals that modifications to the graph are necessary after a change to a certain edge
+    enum EdgeIs: Error {
+        case absent
+        case saturated
+        case oversaturated
+    }
+    
+    /// Increases the flow through the given `edge` by `amount`. Throws `edgeAbsent` if `edge`
+    /// is not found in the network's underlying graph, and `capacityExceeded` if the amount
+    /// being added exceeds the remaining capacity of the edge.
+    mutating func increaseFlow(through edge: Edge, by amount: Weight) throws {
```

Comment:
An alternative to throwing here could be to return an enum with the status of the `Edge`:

```Swift
func increaseFlow(through edge: Edge, by amount: Weight) -> EdgeIs { ... }
```

Reply:
It might be nice to decouple the `.absent` case from the saturation cases. For example, if edge absent-ness was an actual throwable error, while the saturation state could be encoded into an `enum` which is handled as a return value.

The `.absent` case seems much more general (belonging to `Graph`, rather than `FlowNetwork`), while saturation is a `FlowNetwork`-specific attribute.

Reply:
Definitely!